Gems/Terrain/Code/Tests/TerrainPhysicsColliderTests.cpp

```diff
@@ -502,3 +502,95 @@ TEST_F(TerrainPhysicsColliderComponentTest, TerrainPhysicsColliderGetHeightsAndM
 
     m_entity.reset();
 }
+
+TEST_F(TerrainPhysicsColliderComponentTest, TerrainPhysicsColliderDefaultMaterialAssignedWhenTagHasNoMapping)
+{
+    CreateEntity();
+
+    m_boxComponent = m_entity->CreateComponent<UnitTest::MockAxisAlignedBoxShapeComponent>();
+    m_app.RegisterComponentDescriptor(m_boxComponent->CreateDescriptor());
+
+    // Create two SurfaceTag/Material mappings and add them to the collider.
+    Terrain::TerrainPhysicsColliderConfig config;
+
+    const Physics::MaterialId defaultSurfaceMaterial = Physics::MaterialId::Create();
+    const Physics::MaterialId mat1 = Physics::MaterialId::Create();
+
+    const SurfaceData::SurfaceTag tag1 = SurfaceData::SurfaceTag("tag1");
+    const SurfaceData::SurfaceTag tag2 = SurfaceData::SurfaceTag("tag2");
+
+    Terrain::TerrainPhysicsSurfaceMaterialMapping mapping1;
+    mapping1.m_materialId = mat1;
+    mapping1.m_surfaceTag = tag1;
+    config.m_surfaceMaterialMappings.emplace_back(mapping1);
+    config.m_defaultMaterialSelection.SetMaterialId(defaultSurfaceMaterial);
```

Comment:
Suggestion: Might want to make a second similar test where there aren't _any_ mappings set up, to verify there aren't any hidden dependencies where at least one mapped material needs to exist.

Reply:
Suggestion: It'd be good to have another test where default surface material in `TerrainPhysicsColliderConfig` is not set, to check nothing breaks if the user doesn't set a default material to the component.

Reply:
@moraaar This is already tested in the cases above (i.e. the existing tests). 
@mbalfour-amzn Yep, good idea! Adding that